### PR TITLE
feat(daypicker): Initial implementation of controlled daypicker compo…

### PR DESCRIPTION
**Pull request by rachmulvey**
…nent. Abstracted shared parts of popupfield into shared folder

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/35)

### CODE REVIEW

File: src/components/DayPicker/components/Day.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        aria-label={`${dayStartAriaLabel}, ${dayEndAriaLabel}`}
+      >
+        <div css={dateStyles(modifiers)}>{date}</div>
+        {modifiers.start && (
+          <SelectionLabel isStartDate label={startLabel} Icon={Icon} />
+        )}
+        {modifiers.end && (
+          <SelectionLabel isStartDate={false} label={endLabel} Icon={Icon} />
+        )}
+        {/* <span
```
Comment (Pallandor) [inline src/components/DayPicker/components/Day.js]: Remove comments

File: src/components/DayPicker/index.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+import ButtonWithDialog, {
+  ButtonContent,
+  transitionStylesSlideUp,
+  dialogStylesFullScreen
+} from '../../shared/ButtonWithDialog';
+import IconCalendar from '../../icons/CalendarIcon';
+import Calendar from './components/Calendar';
+import Header from './components/Header';
+
+const rowStyles = {
```
[the remark this answers is not in the document]
Comment (rachmulvey) [reply src/components/DayPicker/index.js]: Awesome thanks for the info Roger, the scope of this one is just for mobile I'm doing desktop design next and will make sure to test it out on IE11 before making a PR

File: src/components/DayPicker/components/Day.js (hunk 1 above)
Comment (rachmulvey) [reply src/components/DayPicker/components/Day.js]: removed thx

**Pull request merged**